docs/book/src/reference/creating-events.md

```diff
@@ -0,0 +1,109 @@
+# Creating Events
+
+It is often useful to publish *Event* objects from the controller Reconcile function. 
+
+Events allow users to see what is going on with a particular object, and allow automated processes to see and respond to them.
+
+ Recent Events for an object can be viewed by running `kubectl describe <resource kind> <resource name>`.
+ 
+  ### how to implement the solution to raise an event:
+
+a) You need to pass the recorder when you set up the recorder for the controller when the event will be raised see: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4-with-deploy-image/main.go#L103
```

Comment:
```suggestion
### Passing the Recorder to the Controller

To raise 

```

Reply:
Lets first change the controller; otherwise, users will check the compiler failing. 
We can also do a generic example and put the imports.